lib/linguist/strategy/extension.rb

```diff
@@ -27,7 +25,17 @@ def self.call(blob, candidates)
 
       # Public: Return true if filename uses a generic extension.
       def self.generic?(filename)
-        GENERIC.any? { |ext| filename.downcase.end_with? ext }
+        self.load()
```

Comment:
Nit: We don't need the `()` here:

```suggestion
        self.load
```